code-challenges.test.js

```diff
@@ -12,46 +12,133 @@
+  }
+} 
+
+
+
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+describe("tally", () => {
+
+  const votes1 = { upVotes: 13, downVotes: 2 }
+  // Expected output: 11
+  
+  const votes2 = { upVotes: 2, downVotes: 33 }
+  // Expected output: -31
+
+  it("takes in an object that contains up votes and down votes and returns the end tally", () => {
+    expect(tally(votes1)).toEqual(11)
+    expect(tally(votes2)).toEqual(-31)
+  })
+})
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
 
+// INPUT: object 
+// FUNCTION_NAME: tally
+// OUTPUT: number
+// PROCESS:
+// 1) create a function named tally
+// 2) return the key:value1-key:value2
+
+const tally = (object) => { 
+  return object.upVotes - object.downVotes  
+}
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe ("takeOutTheDuplex", () => {
 
-const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+  const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+  const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+  // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
+  it("returns one array with no duplicate values", () => {
+    expect(takeOutTheDuplex(...dataTypesArray1, ...dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+
+// INPUT: two arrays concatinated with spread operator
+// FUNCTION_NAME: takeOutTheDuplex with a parameter ...duplex
+// OUTPUT: one array with out duplicated values
+// PROCESS:
+// 1) create a function and pass in a parameter with spread operator to take in and concatinate 2 arrays
+// 2) filter the parameter by creating an anonymous function and passing only 2 out of 3 possible parameter(value, index, no array needed here)
+// 3) return duplex parameter and using method indexOf compare it and output array without the equated values
+
+// const duplicatesIncluded = dataTypesArray1.concat(dataTypesArray2)
+// console.log(duplicatesIncluded);
+const takeOutTheDuplex = (...duplex) => {
+    return duplex.filter((value, index) => {
+        return duplex.indexOf(value) == index
```

Comment:
Very close. The main problem here is figuring out what duplex is. Try to console.log duplex to see what you are getting, then you can troubleshoot from there.

Reply:
i did consol.log and JEST. everything passed. But can i work on it a little and submit it tomorrow?